src/rdkafka_int.h

```diff
@@ -690,7 +695,8 @@ rd_kafka_max_poll_exceeded (rd_kafka_t *rk) {
  */
 static RD_INLINE RD_UNUSED void
 rd_kafka_app_poll_blocking (rd_kafka_t *rk) {
-        rd_atomic64_set(&rk->rk_ts_last_poll, INT64_MAX);
+        if (rk->rk_type == RD_KAFKA_CONSUMER)
```

Comment:
seems fine, though i wonder if it's moderately unpleasant smell related to having all client types share `rd_kafka_t`.